restapi/src/main/java/io/stargate/web/resources/v2/schemas/UdtsResource.java

```diff
@@ -0,0 +1,380 @@
+
+import static io.stargate.web.docsapi.resources.RequestToHeadersMapper.getAllHeaders;
+
+import com.codahale.metrics.annotation.Timed;
+import com.google.common.collect.Maps;
+import io.stargate.auth.Scope;
+import io.stargate.auth.SourceAPI;
+import io.stargate.auth.entity.ResourceKind;
+import io.stargate.db.schema.Column;
+import io.stargate.db.schema.ImmutableUserDefinedType;
+import io.stargate.db.schema.Keyspace;
+import io.stargate.db.schema.UserDefinedType;
+import io.stargate.web.models.Error;
+import io.stargate.web.models.GetResponseWrapper;
+import io.stargate.web.models.UdtAdd;
+import io.stargate.web.resources.AuthenticatedDB;
+import io.stargate.web.resources.Converters;
+import io.stargate.web.resources.Db;
+import io.stargate.web.resources.RequestHandler;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiParam;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.HeaderParam;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import org.apache.cassandra.stargate.db.ConsistencyLevel;
+
+@Api(
+    produces = MediaType.APPLICATION_JSON,
+    consumes = MediaType.APPLICATION_JSON,
+    tags = {"schemas"})
+@Path("/v2/schemas/keyspaces/{keyspaceName}/types")
+@Produces(MediaType.APPLICATION_JSON)
+public class UdtsResource {
+
+  @Inject private Db db;
+
+  @Timed
+  @GET
+  @ApiOperation(
+      value = "Get all Types",
+      notes = "Get all Types from a given keyspace",
+      response = GetResponseWrapper.class,
+      responseContainer = "List")
+  @ApiResponses(
+      value = {
+        @ApiResponse(code = 200, message = "OK", response = GetResponseWrapper.class),
+        @ApiResponse(code = 400, message = "Bad Request", response = Error.class),
+        @ApiResponse(code = 401, message = "Unauthorized", response = Error.class),
+        @ApiResponse(code = 500, message = "Internal server error", response = Error.class)
+      })
+  public Response listAllUdts(
+      @ApiParam(
+              value =
+                  "The token returned from the authorization endpoint. Use this token in each request.",
+              required = true)
+          @HeaderParam("X-Cassandra-Token")
+          String token,
+      @ApiParam(value = "Name of the keyspace to use for the request.", required = true)
+          @PathParam("keyspaceName")
+          final String keyspaceName,
```

Comment:
Can we also add the `raw` param to all of these to be consistent with the other endpoints?